src/heap.c

```diff
@@ -34,13 +34,20 @@ static void *defaultAlignedAlloc(void *data, size_t alignment, size_t size)
     return aligned_alloc(alignment, size);
 }
 
+static void defaultAlignedFree(void *data, size_t alignment, void *ptr)
+{
+    (void)alignment;
```

Comment:
Can you add a short comment explaining why this cast to `(void)` here exists, please?

Reply:
The cast to (void) is simply to silence warnings due to unused variables.

We might want to introduce something like:

```
#define RAFT_UNUSED __attribute__((__unused__))
```

and then:

```
static void defaultAlignedFree(void *data, RAFT_UNUSED size_t alignment, void *ptr)
{
```

however that'd be probably compiler-specific and might vary across platforms.